### PR TITLE
Conversion of Name to Id

### PR DESCRIPTION
**Pull request by ksridharbabuus**
Converting name index to Id index for better uniqueness of the Organizations, services and type repos. Added Organization Name and updated few methods to refer by Id instead of Name.

### CODE REVIEW

**Comment by ksridharbabuus** [comment]
@astroseger You can also start the changes in the CLI based on the Registry Function prototype changes. Mostly the following changes are required:

1. orgName -> orgId
2. serviceName -> serviceId
3. typeRepoName -> typeRepoId
4. Method Name change for:
getOrganizationByName to getOrganizationById
getServiceRegistrationByName to getServiceRegistrationById
getTypeRepositoryByName to getTypeRepositoryById
Also in this methods the return variable name is also changed to id. Should not impact on the Client Code.
5. Add orgName to createOrganization function call.

**Comment by astroseger** [comment]
@ksridharbabuus 
1. I'm strongly against including "build" folder to github. It is like uploading "exe" files. It is misusing git. It looks like very bad. And it is completely unnecessary.  We have github releases and npm releases for storing all these abi files (including network files). see #53 .

      * The algorithm is following - we deploy contracts to kovan (And to mainnent, when the time comes), and we make github release and npm release. We already did it for https://github.com/singnet/platform-contracts/releases/tag/v0.2.4 with @Dinesh2521. And it works fine.  Both snet-cli and deaemon get abi and networks from releases. 

2. Probably we should have function to change organization name (only owner should be able to do so)

**Comment by ksridharbabuus** [comment]
@astroseger I dont like to keep Build folder in the git. It should be part of git ignore, mostly it might have missed. Will remove it. Thanks!

Will add the function to update orgName as we dont have any method to modify the same. Will do it. Thanks!

**Comment by astroseger** [comment]
@ksridharbabuus 
OrgName should be string or at least bytes (not bytes32)

**Comment by ksridharbabuus** [comment]
will do the change

**Comment by ksridharbabuus** [comment]
@astroseger Changes to Registry are done.

**Comment by astroseger** [comment]
But why bytes?  I've already change snet-cli to deal with orgName type=string....

**Comment by astroseger** [comment]
Actually it is not very important... But I think string makes more sense for orgName to avoid encoding and decoding to and from utf-8.

**Comment by ksridharbabuus** [comment]
@astroseger I prefer to use bytes compared to string in Solidity. I agree there might be additional function call in f/w to decode the value. Let us keep it as bytes.

**Comment by ksridharbabuus** [comment]
@astroseger In case if it takes longer time for the CLI changes (String to bytes) then I can convert to String in the Registry Contract. Let me know. Thanks!

**Comment by astroseger** [comment]
@ksridharbabuus  disadvantage of keeping it in "bytes" is following: we have to agree in which encoding we store organizationName.  And we need to consistently follow this agreement in all apps.  

If we keep it in "string" then this "agreement" is already fixed in solidity, and we don't need to deal with it. So it is a little bit more clean. But of course difference is not so important. @vforvalerio87 @tiero What do you think?

(from other side I chose bytes for metadataURI, because in the future it might be something more complicated than simple utf-8 encoded string. For example it could be http URI + binary hash)

**Comment by ksridharbabuus** [comment]
@astroseger I dont want restrict to UTF8 for the Orgname to enable other languages as well. If we are fine only to have UTF8 then I can convert to string.

**Comment by ksridharbabuus** [comment]
@astroseger anyway for globalization we need to change in lots of other places. Let me change it string then.

File: test/Registry_metadata_test1.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 "use strict";
 var Registry = artifacts.require("./Registry.sol");
+let { HELPERS } = require("./util/Util.js");
+let { bytesToString} = HELPERS;
```
Comment (astroseger) [inline test/Registry_metadata_test1.js]: You don't need bytesToString anymore.. But it is a minor comment anyway...

**Pull request merged**